src/pybind/mgr/cephadm/serve.py

```diff
@@ -1091,10 +1091,11 @@ def _check_daemons(self) -> None:
                 self.log.debug(f'{dd.name()} deps {last_deps} -> {deps}')
                 self.log.info(f'Reconfiguring {dd.name()} (dependencies changed)...')
                 action = 'reconfig'
-                # we need only redeploy if secure_monitoring_stack value has changed:
+                # we need only redeploy if secure_monitoring_stack or mgmt-gateway value has changed:
+                # TODO(redo): check if we should just go always with redeploy (it's fast enough)
                 if dd.daemon_type in ['prometheus', 'node-exporter', 'alertmanager']:
-                    diff = list(set(last_deps) - set(deps))
-                    if any('secure_monitoring_stack' in e for e in diff):
+                    diff = list(set(last_deps).symmetric_difference(set(deps)))
```

Comment:
This is quite a change in behavior... Looks like the previous setting was buggy, or this is overachieving?

Reply:
Good catch!! In fact this was something broken from previous code and I fixed it. I'll be separating this small change on a different PR so we can can back-port it to old code and to make it clear this fixing an issue.

Reply:
agreed in the standup this would become its own PR so we can backport it farther than squid a bit easier

Reply:
@adk3798 I moved the related changes to https://github.com/ceph/ceph/pull/58230